nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/indexing/conditions/Conditions.java

```diff
@@ -24,6 +24,59 @@
 
 public class Conditions {
 
+    private Conditions() {}
+
+    public enum ConditionMode {
+        EPSILON_EQUALS(0), //0
+        EPSILON_NOT_EQUALS(1), //1
+        LESS_THAN(2), //2
+        GREATER_THAN(3), //3
+        LESS_THAN_OR_EQUAL(4), //4
+        GREATER_THAN_OR_EQUAL(5), //5
+        ABS_LESS_THAN(6), //6
+        ABS_GREATER_THAN(7), //7
+        IS_INFINITE(8), //8
+        IS_NAN(9), //9
+        ABS_EQUALS(10), //10
+        NOT_EQUALS(11), //11
+        ABS_GREATER_OR_EQUAL(12), //12
+        ABS_LESS_THAN_OR_EQUAL(13), //13
+        IS_FINITE(14), //14
+        NOT_FINITE(15), //15
+        AGGREGATE(-1); // this is an aggregate enum for or, and, not, and other indirect conditions that depend on others
```

Comment:
No need to have the value commented after it anymore.

Reply:
I actually disagree here and will keep it. The "aggregate" is a catch all for anything that's not a direct evaluation.

Reply:
I'm taking about all the things above it.